conanfile.py

```diff
@@ -136,6 +137,22 @@ def _gather_licenses(self, bootstrap):
             f" --with-license-file --no-license-path --output-file=package_licenses.txt"
         )
 
+    def _gather_packages(self):
+        """Gather all the required packages into a file for future reference
+
+        We don't have an easy way to get this so parse the package names out of the requirements
+        file.
+        """
+        requirements = self.make_requirements_file()
+        packages = pathlib.Path("packages.txt").resolve()
+        matcher = re.compile(r"^([\w.-]+)==[\w.-]+$", flags=re.MULTILINE)
+
+        with open(requirements, "r") as input, open(packages, "w") as output:
+            for entry in input:
+                if match := matcher.match(entry):
+                    output.write(match.group(1) + "\n")
```

Comment:
`.make_requirements_file()` does two things: it creates a `package_list` as a regular Python list, and then it writes that to a file as `"\n".join(package_list)`. It would be useful to factor out a `.make_package_list()` function out and make use of it both here and in `.make_requirements_file()`. Then there's less uncertainty about the format.
```suggestion
        matcher = re.compile(r"^([\w.-]+)==[\w.-]+$")
        matches = map(matcher.match, self.make_package_list())
        package_names = (match.group(1) for match in filter(None, matches))
        with open("packages.txt", "w") as output:                    
            output.write("\n".join(package_names))
```